### DISCUSSION TITLE
Provide more visibility into filtered versions

### DISCUSSION THREAD

**Discussion by rarkins** [renovatebot/renovate#29152] [Suggest an Idea]
### Tell us more.

There are now many reasons why a "newer" version might be ignored or suppressed, including:
- `allowedVersions` and `minimumReleaseAge` configuration
- `constraintsFiltering=strict`
- Various `compatibility` filtering
- Unstable/non-LTS (e.g. Node)

I don't want to make the information "in your face" (demanding users look at it) but I do want to make it visible at least in logs for those who are interested or troubleshooting.

Generally speaking our approach to filtering is that the initial list we get from a datasource lookup gets progressively narrowed before eventually being grouped into "buckets" (e.g. major, non-major). If a version is filtered out for more than one reason (e.g it could be unstable _and_ not satisfying minimumReleaseAge) then we don't check it for every reason once it's already failed one.

In some cases like npm packages or the Node.js version it would make a lot of sense to have full visibility into which reason any version was filtered out. For other cases like Docker tags - including Node.js docker tags - I'm not sure we want to list every single version from a wrong "compatibility" (e.g. -alpine -debian -slim, etc) as it can be too much information.

**Comment by secustor** [comment]
Makes sense. 

On DEBUG we should log how many releases have been discarded by reason e.g. `unstable` and on TRACE a complete list of releases with their specific reason.

**Comment by emmercm** [comment]
> I don't want to make the information "in your face" (demanding users look at it)

Maybe a good way to surface the information would be the project's dependency dashboard, if enabled. I have a Node.js/npm project with several dependencies being held back by my `package.json`'s `engine.node` constraint, with this Renovate config:

```json5
{
  packageRules: [
    {
      matchDepTypes: ['rollback'],
      schedule: 'at any time'
    }
  ],
  npm: {
    packageRules: [
      {
        matchDepTypes: ['dependencies'],
        constraintsFiltering: 'strict',
        rollbackPrs: true
      }
    ]
  }
}
```

I have the dependency dashboard enabled, and I would love to summarized there what dependencies are being held back, which would help inform a possible change to `engine.node`.
